pkg/rke2/kp.go

```diff
@@ -0,0 +1,43 @@
+package rke2
+
+import (
+	"context"
+
+	"github.com/rancher/k3s/pkg/cli/cmds"
+	"github.com/rancher/k3s/pkg/server"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const kubeProxyChart = "rke2-kube-proxy"
+
+// setKubeProxyDisabled determines if a cluster already has kube proxy deployed as a chart, if so
+// disables running kubeproxy as a static pod.
+func setKubeProxyDisabled(clx *cli.Context, cfg *cmds.Server) func(context.Context, <-chan struct{}, string) error {
+	return func(ctx context.Context, apiServerReady <-chan struct{}, kubeConfigAdmin string) error {
+		go func() {
+			<-apiServerReady
+			logrus.Info("Checking for kube proxy as a chart")
+
+			sc, err := server.NewContext(ctx, kubeConfigAdmin)
+			if err != nil {
+				logrus.Fatalf("KubeProxyChartCheck: new server context: %s", err.Error())
+			}
+
+			helmChartConfig, err := sc.Helm.Helm().V1().HelmChartConfig().Get(metav1.NamespaceSystem, kubeProxyChart, metav1.GetOptions{})
+			if err != nil {
+				logrus.WithError(err).Info("Kube Proxy chart config wasn't found")
+			}
+			if helmChartConfig != nil {
+				cfg.DisableKubeProxy = true
+				if err := clx.Set("disable", kubeProxyChart); err != nil {
+					logrus.Fatalf("KubeProxyChartCheck: disabling the chart: %s", err.Error())
+				}
+
+			}
+			logrus.Info("Kube proxy chart check completed")
```

Comment:
* Don't use server.NewContext, it sets up a lot more than we need. 
* Need to handle NotFound errors specifically, as that's not the only kind of issue we might get back.
* If the config IS found, we need to `--disable-kube-proxy`, if it IS NOT found we need to `--disable=rke2-kube-proxy`

Something like this perhaps:

```suggestion
			restConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigAdmin)
			if err != nil {
				logrus.Fatalf("kp: new k8s client: %s", err.Error())
			}
			
			hc, err := helm.NewForConfig(restConfig)
			if err != nil {
				logrus.Fatalf("kp: new helm client: %s", err.Error())
			}

			if _, err := hc.HelmV1().HelmChartConfig().Get(metav1.NamespaceSystem, kubeProxyChart, metav1.GetOptions{}); err != nil {
				if apierrors.IsNotFound(err) {
					logrus.Infof("%[1]s HelmChartConfig not found, disabling %[1]s", kubeProxyChart)
					return clx.Set("disable", kubeProxyChart);
				} else {
					return errors.Wrap(err, )
				}
			}
			logrus.Infof("%s HelmChartConfig found, disabling embedded kube-proxy", kubeProxyChart)
			cfg.DisableKubeProxy = true
			return nil
			
```